transcoder/output/json/JsonOutputManager.py

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright 2022 Google LLC
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import os
+import sys
+
+#from genson import SchemaBuilder
+from transcoder.message import DatacastSchema, DatacastField
+from transcoder.output import OutputManager
+
+class JsonOutputManager(OutputManager):
+    """Transcode messages to JSON encoding persisted to a file"""
+    def __init__(self, prefix: str, output_path: str, lazy_create_resources: bool = False):
+        super().__init__(lazy_create_resources=lazy_create_resources)
+        self.prefix = prefix
+        self.schemas = {}
+        self.writers = {}
+
+        if output_path is None:
+            rel_path = "jsonOut"
+            main_script_dir = os.path.dirname(sys.argv[0])
+            self.output_path = os.path.join(main_script_dir, rel_path)
+        else:
+            self.output_path = output_path
+
+        exists = os.path.exists(self.output_path)
+        if not exists:
+            os.makedirs(self.output_path)
+
+    @staticmethod
+    def init_schema():
+        schema = {}
+        schema['$schema'] = 'https://json-schema.org/draft/2019-09/schema'
+        schema['type'] = 'object'
+        return schema
+
+    @staticmethod
+    def output_type_identifier():
+        return 'json'
```

Comment:
Should this be "jsonl"?

Reply:
so that's a good point and I would say technically correct but in terms of common usage I think `json` is a bit more intuitive out of the gate than `jsonl` hence why it was picked. I don't have a strong feeling about it either way really

Reply:
thinking about this more I think it is better to be more precise, will update CL